Assets/Scripts/Visuals/UiHandler/RightClickHandler.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Assets.Scripts.Visuals.UiHandler
+{
+    public class RightClickHandler : MonoBehaviour, IPointerDownHandler
+    {
+        public Action<Vector3> rightClick;
```

Comment:
Name action callbacks so that it is obvious that they are callbacks. Something like `onRightClick` or `rightClickCallback`